test/e2e.sh

```diff
@@ -0,0 +1,130 @@
+WEBUI_TEST_EXTERNAL_USER=${WEBUI_TEST_EXTERNAL_USER:-}
+WEBUI_TEST_EXTERNAL_PASSWORD=${WEBUI_TEST_EXTERNAL_PASSWORD:-}
+WEBUI_TEST_RESOURCES=${WEBUI_TEST_RESOURCES:-$TOP_DIR/tools/resources.yaml}
+if [ -z "$WEBUI_TEST_IMAGE" ]; then
+    if [ "$WEBUI_TEST_LOCAL_IMAGE" == true ]; then
+        WEBUI_TEST_IMAGE=oshinko-webui:latest
+    else
+        WEBUI_TEST_IMAGE=docker.io/radanalyticsio/oshinko-webui
+    fi
+fi
+
+function print_test_env {
+    echo Using image $WEBUI_TEST_IMAGE
+    echo Using resources.yaml from $WEBUI_TEST_RESOURCES
+
+    if [ "$WEBUI_TEST_LOCAL_IMAGE" != true ]; then
+	echo WEBUI_TEST_LOCAL_IMAGE = $WEBUI_TEST_LOCAL_IMAGE, webui image is external, ignoring registry env vars
+    elif [ -n "$WEBUI_TEST_EXTERNAL_REGISTRY" ]; then
+        echo Using external registry $WEBUI_TEST_EXTERNAL_REGISTRY
+        if [ -z "$WEBUI_TEST_EXTERNAL_USER" ]; then
+            echo "Error: WEBUI_TEST_EXTERNAL_USER not set!"
+	    exit 1
+        else
+	    echo Using external registry user $WEBUI_TEST_EXTERNAL_USER
+        fi
+        if [ -z "$WEBUI_TEST_EXTERNAL_PASSWORD" ]; then
+            echo "WEBUI_TEST_EXTERNAL_PASSWORD not set, assuming current docker login"
+        else
+            echo External registry password set
+        fi
+    elif [ -n "$WEBUI_TEST_INTEGRATED_REGISTRY" ]; then
+        echo Using integrated registry $WEBUI_TEST_INTEGRATED_REGISTRY
+    else
+        echo Not using external or integrated registry
+    fi
+}
+
+function push_image {
+    # The ip address of an internal/external registry may be set to support running against
+    # an openshift that is not "oc cluster up" when using images that have been built locally.
+    # In the case of "oc cluster up", the docker on the host is available from openshift so
+    # no special pushes of images have to be done.
+    # In the case of a "normal" openshift cluster, a local image we'll use for build has to be
+    # available from the designated registry.
+    # If we're using a pyspark image already in an external registry, openshift can pull it from
+    # there and we don't have to do anything.
+    local user=
+    local password=
+    local pushproj=
+    local pushimage=
+    local registry=
+    if [ "$WEBUI_TEST_LOCAL_IMAGE" == true ]; then
+	if [ -n  "$WEBUI_TEST_EXTERNAL_REGISTRY" ]; then
+	    user=$WEBUI_TEST_EXTERNAL_USER
+	    password=$WEBUI_TEST_EXTERNAL_PASSWORD
+	    pushproj=$user
+	    pushimage=scratch-oshinko-webui
+	    registry=$WEBUI_TEST_EXTERNAL_REGISTRY
+	elif [ -n "$WEBUI_TEST_INTEGRATED_REGISTRY" ]; then
+	    user=$(oc whoami)
+	    password=$(oc whoami -t)
+	    pushproj=$PROJECT
+	    pushimage=oshinko-webui
+	    registry=$WEBUI_TEST_INTEGRATED_REGISTRY
+	fi
+    fi
+    if [ -n "$registry" ]; then
+	set +e
+	docker login --help | grep email &> /dev/null
+	res=$?
+	set -e
+	if [ -n "$password" ]; then
+	    if [ "$res" -eq 0 ]; then
+		docker login -u ${user} -e jack@jack.com -p ${password} ${registry}
```

Comment:
should be able to remove the -e <email>

Reply:
I had this issue somewhere, different versions of docker. Maybe that's past now.

Reply:
yeah, this is necessary for origin 1.5, different version of docker. I'm on the fence about leaving it, the s2i tests have it.